src/dynamixel/controllers/usb2dynamixel.hpp

```diff
@@ -105,45 +107,74 @@ namespace dynamixel {
+
                 if (_fd == -1)
                     return false;
 
                 double time = get_time();
-                bool done = false;
+                DecodeState state = DecodeState::ONGOING;
 
                 std::vector<uint8_t> packet;
                 packet.reserve(_recv_buffer_size);
 
-                //std::cout << "Recv: ";
+                //std::cout << "Receive:" << std::endl;
 
                 do {
                     double current_time = get_time();
-                    uint8_t b;
-                    int res = read(_fd, &b, 1);
+                    uint8_t byte;
+                    int res = read(_fd, &byte, 1);
                     if (res > 0) {
-                        packet.push_back(b);
-                        time = current_time;
+                        // std::cout << std::setfill('0') << std::setw(2)
+                        //           << std::hex << (unsigned int)byte << " ";
+                        packet.push_back(byte);
+
+                        state = status.decode_packet(packet, _report_bad_packet);
+                        if (state == DecodeState::INVALID) {
+                            // std::cout << "\tBad packet: ";
+                            // for (const auto byte : packet)
+                            //     std::cout << std::setfill('0') << std::setw(2) << std::hex << (unsigned int)byte << " ";
+                            // std::cout << std::endl;
 
-                        //  std::cout << "0x" << std::setfill('0') << std::setw(2) << std::hex << (unsigned int)b << " ";
+                            packet.clear();
+                        }
 
-                        done = status.decode_packet(packet);
+                        time = current_time;
                     }
 
-                    if (current_time - time > _recv_timeout) {
-                        //std::cout << std::endl;
+                    if (current_time - time > _recv_timeout)
                         return false;
-                    }
-                } while (!done);
+                } while (state != DecodeState::DONE);
 
                 //std::cout << std::endl;
+                std::cout << std::dec;
 
                 return true;
             }
 
+            /** Enable error reporting for packet issues
+
+                If report_bad_packet is set to true, invalid packet headers and
+                errors in packet size are reported through exceptions.
+            **/
+            void set_report_bad_packet(bool report_bad_packet)
+            {
+                _report_bad_packet = report_bad_packet;
+            }
+
+            /**
+                @see set_report_bad_packet
+            **/
+            bool report_bad_packet()
+            {
+                return _report_bad_packet;
+            }
+
         private:
             double _recv_timeout;
             static const size_t _recv_buffer_size = 256;
             int _fd;
+            bool _report_bad_packet;
```

Comment:
It is only used at line 132, passed to the method that actually processes the packet.